src/tools/w3dviewer/mainfrm.cpp

```diff
@@ -222,11 +224,10 @@ BOOL CMainFrame::OnCommand(WPARAM wParam, LPARAM lParam)
         info.cch = sizeof(name);
 
         if (GetMenuItemInfo(m_subMenu, LOWORD(wParam), FALSE, &info)) {
-            // do later
-            // EditorParticleEmitterDefClas *def = new EditorParticleEmitterDefClass();
-            // GetCurrentDocument()->AddEmittersToDef(def, reinterpret_cast<const char *>(&info.hbmpItem), nullptr);
-            // EmitterPropertySheetClass sheet(def, IDS_EMITTERPROPERTIES, this);
-            // sheet.DoModal();
+            EmitterInstanceList *list = new EmitterInstanceList();
+            GetCurrentDocument()->AddEmittersToList(list, reinterpret_cast<const char *>(&info.hbmpItem), nullptr);
```

Comment:
Why is this casted to `const char*` ? I assume it is a `HBITMAP` ?

Reply:
fixed.